### PR TITLE
Initial IL support

### PR DESCRIPTION
**Pull request by kkokosa**
Adding minimal set of changes that are consumed to plug-in IL into SharpLab:

- `AspNetCore.Demo` with IL tab
- IL session
- MirroSharp.IL NuGet generation

### CODE REVIEW

File: AspNetCore.Demo/AspNetCore.Demo.csproj [hunk 1]
```diff
@@ -20,19 +20,15 @@
...
     <ProjectReference Include="..\FSharp\FSharp.csproj" />
+    <ProjectReference Include="..\IL\IL.csproj" />
   </ItemGroup>
 
   <Target Name="CopySdkReferenceAssemblies" AfterTargets="ResolveReferences">
     <ItemGroup>
-      <Content Include="@(Reference->'%(RootDir)%(Directory)%(Filename)%(Extension)')"
-               Link="@(Reference->'ref-assemblies\%(Filename)%(Extension)')"
-               Condition="$([System.String]::new('%(Filename)').StartsWith('System')) or '%(Filename)' == 'netstandard'">
+      <Content Include="@(Reference->'%(RootDir)%(Directory)%(Filename)%(Extension)')" Link="@(Reference->'ref-assemblies\%(Filename)%(Extension)')" Condition="$([System.String]::new('%(Filename)').StartsWith('System')) or '%(Filename)' == 'netstandard'">
```
Comment (ashmind) [inline AspNetCore.Demo/AspNetCore.Demo.csproj:28]: This formatting change (same below) looks unintentional -- can be reverted?

File: IL/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Changelog
+
```
Comment (ashmind) [inline IL/CHANGELOG.md:2]: This doesn't have to be empty, could do something like 0.1 Initial version. Not a blocker for PR as I can add that.

File: IL/Internal/ILSession.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        public ILSession(string text, MirrorSharpILOptions options) {
+            _text = text;
+            _options = options;
+        }
+
+        public string GetText() => _text;
+
+        public void ReplaceText(string? newText, int start = 0, int? length = null) {
+            if (length > 0)
+                _text = _text.Remove(start, length.Value);
```
Comment (ashmind) [inline IL/Internal/ILSession.cs:28]: I'll need to remember details here, but I think client sometimes sends text char-by-char, wonder if StringBuilder would be more efficient here.

File: IL/MirrorSharpOptionsExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+using System;
+using Microsoft.CodeAnalysis;
+using MirrorSharp.IL;
+using MirrorSharp.IL.Internal;
+using MirrorSharp.Internal;
+
+// ReSharper disable once CheckNamespace
+namespace MirrorSharp
+{
```
Comment (ashmind) [inline IL/MirrorSharpOptionsExtensions.cs:9]: Brace does not follow end-of-line, same below. Not a blocker, I can adjust.

**Comment by ashmind** [comment]
Thanks for submitting this! I'll take another look tomorrow NZ time. All comments are not blocking, if cosmetics are pending by the time I'll just adjust them, so no rush.

File: IL/Internal/ILSession.cs (hunk 3 above)
Comment (kkokosa) [reply IL/Internal/ILSession.cs:28]: Probably! I've just copied code from FSharp version as an initial version.

File: IL/MirrorSharpOptionsExtensions.cs (hunk 4 above)
Comment (kkokosa) [reply IL/MirrorSharpOptionsExtensions.cs]: Fixed!

File: IL/CHANGELOG.md (hunk 2 above)
Comment (kkokosa) [reply IL/CHANGELOG.md:2]: Added.

File: AspNetCore.Demo/AspNetCore.Demo.csproj (hunk 1 above)
Comment (kkokosa) [reply AspNetCore.Demo/AspNetCore.Demo.csproj]: Yep, sorry! Fixed.

**Pull request merged**